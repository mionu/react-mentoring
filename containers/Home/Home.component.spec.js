import React from 'react';
import { RouterContext } from 'next/dist/next-server/lib/router-context';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import HomeComponent from './Home.component';

/* eslint-disable no-undef, react/jsx-filename-extension, react/jsx-props-no-spreading */
describe('HomeComponent', () => {
  beforeAll(() => jest.spyOn(React, 'useEffect').mockImplementation(React.useLayoutEffect));
  afterAll(() => React.useEffect.mockRestore());

  const mockProps = {
    search: jest.fn(),
    loading: false,
    movies: [],
  };
  const mockStore = configureStore([]);

  let store;

  beforeEach(() => {
    store = mockStore({
      movies: {
        list: [],
        options: {},
      },
      shared: {
        loading: {},
      },
    });
  });

  it('should render', () => {
    const tree = renderer
      .create(
        <RouterContext.Provider value={{ query: { query: 'a' } }}>
          <Provider store={store}>
            <HomeComponent {...mockProps} />
          </Provider>
        </RouterContext.Provider>,
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
/* eslint-enable no-undef, react/jsx-filename-extension, react/jsx-props-no-spreading */
