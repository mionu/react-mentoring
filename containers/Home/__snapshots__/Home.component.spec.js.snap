// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent should render 1`] = `
Array [
  <div
    aria-hidden={true}
    className="MuiBackdrop-root makeStyles-backdrop-1"
    style={
      Object {
        "opacity": 0,
        "visibility": "hidden",
      }
    }
  >
    <div
      className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary"
      role="progressbar"
      style={
        Object {
          "height": "5rem",
          "width": "5rem",
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
  </div>,
  <div
    className="makeStyles-header-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
        Add movie
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>,
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <h1>
      FIND YOUR MOVIE
    </h1>
    <div
      className="makeStyles-search-4"
    >
      <div
        className="MuiFormControl-root MuiTextField-root makeStyles-searchInput-5"
        onKeyPress={[Function]}
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
          data-shrink={false}
        >
          What do you want to watch?
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-8"
            >
              <span>
                What do you want to watch?
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Search
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      className="makeStyles-flexSpaceBetween-11"
    >
      <div
        className="MuiTabs-root"
      >
        <div
          className="MuiTabs-scroller MuiTabs-fixed"
          onScroll={[Function]}
          style={
            Object {
              "marginBottom": -0,
              "overflow": "hidden",
            }
          }
        >
          <div
            aria-orientation={null}
            className="MuiTabs-flexContainer"
            onKeyDown={[Function]}
            role="tablist"
          >
            <button
              aria-selected={true}
              className="MuiButtonBase-root MuiTab-root makeStyles-root-15 MuiTab-textColorInherit Mui-selected"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiTab-wrapper"
              >
                all
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected={false}
              className="MuiButtonBase-root MuiTab-root makeStyles-root-15 MuiTab-textColorInherit"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiTab-wrapper"
              >
                documentary
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected={false}
              className="MuiButtonBase-root MuiTab-root makeStyles-root-15 MuiTab-textColorInherit"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiTab-wrapper"
              >
                comedy
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected={false}
              className="MuiButtonBase-root MuiTab-root makeStyles-root-15 MuiTab-textColorInherit"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiTab-wrapper"
              >
                horror
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected={false}
              className="MuiButtonBase-root MuiTab-root makeStyles-root-15 MuiTab-textColorInherit"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiTab-wrapper"
              >
                crime
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
          </div>
          <span
            className="PrivateTabIndicator-root-16 PrivateTabIndicator-colorSecondary-18 MuiTabs-indicator"
            style={
              Object {
                "left": 0,
                "width": 0,
              }
            }
          />
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root makeStyles-select-20 MuiFormControl-marginDense"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled"
          data-shrink={true}
        >
          SORT BY
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          onClick={[Function]}
        >
          <div
            aria-haspopup="listbox"
            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            RELEASE DATE
          </div>
          <input
            aria-hidden={true}
            autoFocus={false}
            className="MuiSelect-nativeInput"
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
            value="release_date"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-8 PrivateNotchedOutline-legendNotched-9"
            >
              <span>
                SORT BY
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="makeStyles-emptyMoviesList-10"
    >
      <h1>
        No movies found
      </h1>
    </div>
  </div>,
]
`;
